psydac/api/ast/nodes.py

```diff
@@ -42,6 +42,20 @@ class Zeros(Function):
     def shape(self):
         return self._args[0]
 
+class FloorDiv(Function):
+    def __new__(cls, a1, a2):
+        if a2 == 1:
+            return a1
+        else:
+            return Basic.__new__(cls, a1, a2)
+
+    @property
+    def a1(self):
+        return self._args[0]
+
+    @property
+    def a2(self):
+        return self._args[1]
```

Comment:
I don't think that `a1` and `a2` are expressive names for these properties. Doesn't `FloorDiv` inherit the `args` property from the `Function` class already?

Reply:
yes, `args` are inherited from the `Function` class,  so I have changed them to `arg1` and `arg2` to be more clear.